modules/core/tests/test_foreach_attribute.cpp

```diff
@@ -463,3 +492,23 @@ TEST_CASE("SurfaceMesh: Foreach CoW", "[next]")
 #define LA_X_test_foreach_cow(_, Scalar, Index) test_foreach_cow<Scalar, Index>();
     LA_SURFACE_MESH_X(test_foreach_cow, 0)
 }
+
+TEST_CASE("SurfaceMesh: Parallel CoW", "[next]")
+{
+    test_parallel_cow<double>();
+}
+
+TEST_CASE("parallel_cow_tbb", "[next]")
+{
+    int N = 100;
+    using Scalar = double;
+
+    std::vector<lagrange::copy_on_write_ptr<ArrayBase>> attrs;
+    for (int i = 0; i < N; ++i) {
+        auto ptr = lagrange::internal::make_shared<Array<Scalar>>();
+        ptr->data.resize(10);
+        attrs.emplace_back(std::move(ptr));
+        attrs.push_back(attrs.back());
+    }
+    tbb::parallel_for(0, N, [&](int i) { attrs[i].static_write<Array<Scalar>>()->data[0] = 1; });
+}
```

Comment:
How does this differ from `test_parallel_cow<double>()` above?

Reply:
Hmm I guess it may have differed in the course of testing, but now it's the same. I'll remove it.